src/operator/tensor/elemwise_binary_broadcast_op.h

```diff
@@ -155,6 +181,22 @@ struct binary_broadcast_kernel {
     }
   }
 };
+
+template<typename DType, typename CType, typename RType, int req, typename OP>
```

Comment:
can the template types be at function level and be inferred automatically by the arguments passed, or are you going for type checking?

Reply:
Agree with applying these as function's template arguments instead of the class's.

Reply:
Will make that change soon.